File: 90-subsets-ii/90-subsets-ii.cpp
class Solution {
public:
    vector<vector<int>> ret;
    
    void solve(vector<int>& nums, vector<int>& v, int idx) {
        
        ret.push_back(v);
        
        for (int i = idx; i < nums.size(); i++) {
            
            if (i > idx && nums[i] == nums[i - 1]) continue;
            
            v.push_back(nums[i]);
            solve(nums, v, i + 1);
            v.pop_back();
        }
    }
    
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        
        sort(nums.begin(), nums.end());
        
        vector<int> v;
        solve(nums, v, 0);
        
        return ret;
    }
};

File: 90-subsets-ii/90-subsets-ii.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        
        answer = []
        numsLen = len(nums)
        nums.sort()
        
        def solve(subArray, curIdx):
            
            answer.append(subArray.copy())
            
            for i in range(curIdx, numsLen):
                
                if i > curIdx and nums[i] == nums[i - 1]:
                    continue
                
                subArray.append(nums[i])
                solve(subArray, i + 1)
                subArray.pop()
                
        solve([], 0)
        
        return answer